drivers/misc/coresight/nrf_etr.c

```diff
@@ -0,0 +1,691 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/cache.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_output.h>
+#include <zephyr/logging/log_frontend_stmesp.h>
+#include <zephyr/logging/log_frontend_stmesp_demux.h>
+#include <zephyr/debug/coresight/cs_trace_defmt.h>
+#include <zephyr/debug/mipi_stp_decoder.h>
+#include <zephyr/linker/devicetree_regions.h>
+#include <zephyr/drivers/misc/coresight/nrf_etr.h>
+#include <zephyr/sys/printk.h>
+#include <nrfx_tbm.h>
+#include <stdio.h>
+LOG_MODULE_REGISTER(cs_etr_tbm);
+
+#define UART_NODE DT_CHOSEN(zephyr_console)
```

Comment:
The console is not necessarily a uart, is it possible to check that it actually is a uart? 

Also would it be possible in the future to just use the stdout registered by the console driver if there is a zephyr_console? So transport could also be segger rtt, semihosting and whatnot?

Reply:
There are multiple consoles types but ias for the chosen `zephyr_console` i think that it is always an uart. I agree that In future we might extend transport.